gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/SearchTopHitsResponseAdapter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.repository.elasticsearch.v4.analytics.adapter;
+
+import static io.gravitee.repository.elasticsearch.v4.analytics.adapter.SearchTopHitsQueryAdapter.API_TOP_HITS_COUNT;
+import static io.gravitee.repository.elasticsearch.v4.analytics.adapter.SearchTopHitsQueryAdapter.HITS_COUNT;
```

Comment:
For response status and response time at API-level we merge the adaptor for query and for response because you have a big coupling between 2.

I think it can be useful here too.